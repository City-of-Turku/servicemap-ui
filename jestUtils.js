import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });