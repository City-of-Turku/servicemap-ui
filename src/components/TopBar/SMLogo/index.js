import { withStyles } from '@mui/styles';
import styles from '../styles';
import SMLogoComponent from './SMLogoComponent';

const SMLogo = withStyles(styles)(SMLogoComponent);

export default SMLogo;
