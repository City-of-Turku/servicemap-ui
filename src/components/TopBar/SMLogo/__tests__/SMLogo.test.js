// Link.react.test.js
import React from 'react';
import { fireEvent } from '@testing-library/react';
import SMLogo from '../index';
import { initialState } from '../../../../redux/reducers/user';
import { getRenderWithProviders } from '../../../../../jestUtils';
import finnishTranslations from '../../../../i18n/fi';

const renderWithProviders = getRenderWithProviders({
  user: initialState,
  settings: {},
});

describe('<SMLogo />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<SMLogo />);
    expect(container).toMatchSnapshot();
  });

  it('simulates click event', () => {
    const mockCallBack = jest.fn();
    const { container } = renderWithProviders(
      <SMLogo
        onClick={mockCallBack}
      />,
    );
    fireEvent.click(container.querySelector('button'));
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it('does render default accessibility attributes correctly', () => {
    const { container } = renderWithProviders(
      <SMLogo onClick={() => {}} />,
    );
    const buttonBase = container.querySelector('button');
    // Expect role to be link
    expect(buttonBase).toHaveAttribute('role', 'link');
    // Home logo which is image should containe image text in aria label
    // so aria label should contain "Palvelukartta" text in addition to action description
    expect(buttonBase).toHaveAttribute('aria-label', finnishTranslations['general.home.logo.ariaLabel']);
    // Expect visible text to be hidden from screen readers
    expect(buttonBase.querySelector('div')).toHaveAttribute('aria-hidden', 'true');
    expect(buttonBase.querySelector('div')).toHaveAttribute('role', 'img');
  });
});
