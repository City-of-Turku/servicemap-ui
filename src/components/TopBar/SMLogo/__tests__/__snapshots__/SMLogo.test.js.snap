// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SMLogo /> should work 1`] = `
<div>
  <button
    aria-label="Palvelukartta - Siirry etusivulle"
    class="MuiButtonBase-root css-5x6wf3-MuiButtonBase-root"
    role="link"
    tabindex="0"
    type="button"
  >
    <div
      aria-hidden="true"
      class="SMLogoComponent-logo-6"
      role="img"
    >
      <img
        alt=""
        class="HomeLogo-icon-26"
        src="test-file-stub"
      />
    </div>
  </button>
</div>
`;
